security/advancedtls/advancedtls_integration_test.go

```diff
@@ -784,7 +785,7 @@ func (s) TestDefaultHostNameCheck(t *testing.T) {
 				t.Fatalf("failed to listen: %v", err)
 			}
 			defer lis.Close()
-			addr := lis.Addr().String()
+			addr := strings.Replace(lis.Addr().String(), "127.0.0.1", "localhost", 1)
```

Comment:
This won't work with IPv6.

How about this:

```go
addr := fmt.Sprintf("localhost:%v", lis.Addr().(*net.TCPAddr).Port)
```

Reply:
Yeah I did neglect that. Thanks for the suggestion! It should all work now. Would you please give another look?